src/tools/imgupload/imguploadermanager.cpp

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: Yurii Puchkov & Contributors
+//
+
+#include "imguploadermanager.h"
+#include <QPixmap>
+#include <QWidget>
+
+// TODO - remove this hard-code and create plugin manager in the future, you may
+// include other storage headers here
+#include "storages/imgur/imguruploader.h"
+
+ImgUploaderManager::ImgUploaderManager(QObject* parent)
+  : QObject(parent)
+  , m_imgUploaderBase(nullptr)
+{
+    // TODO - implement ImgUploader for other Storages and selection among them
+    m_imgUploaderPlugin = IMG_UPLOADER_STORAGE_DEFAULT;
+    init();
+}
+
+void ImgUploaderManager::init()
+{
+    // TODO - implement ImgUploader for other Storages and selection among them,
+    // example:
+    // if (uploaderPlugin().compare("s3") == 0) {
+    //    m_qstrUrl = ImgS3Settings().value("S3", "S3_URL").toString();
+    //} else {
+    //    m_qstrUrl = "https://imgur.com/";
+    //    m_imgUploaderPlugin = "imgur";
+    //}
+    m_urlString = "https://imgur.com/";
+    m_imgUploaderPlugin = "imgur";
+}
+
+ImgUploaderBase* ImgUploaderManager::uploader(const QPixmap& capture,
+                                              QWidget* parent)
+{
+    // TODO - implement ImgUploader for other Storages and selection among them,
```

Comment:
This TODO does not reflect how we will implement more uploaders. Rather than being hardcoded in the c++ source we will parse config files.

Reply:
Probably "TODO" is not a perfect marker here, sorry for confusing.

It works for both options:
- implement plugin system here (we'll parse config files later)
- if we won't find a time for that it is also possible to add just C++ classes for other providers here (I did such approach for Namecheap fork).

The main purpose of this PR is to remove "Imgur" code from Capture, History and other classes and move to a single place.

Please merge it if you have no strict objections.